fault-remediation/pkg/reconciler/reconciler.go

```diff
@@ -454,8 +440,10 @@ func (r *FaultRemediationReconciler) checkExistingCRStatus(
 
 	slog.Info("CR completed or failed, allowing retry", "node", nodeName, "crName", groupState.MaintenanceCR)
 
-	if err := r.annotationManager.RemoveGroupFromState(ctx, nodeName, group); err != nil {
+	//TODO: this also has built in client retry on the api request
```

Comment:
Yeah so it looks like some of the managers like annotationManager have built in retries. I'm planning to have a flag set on the manager struct that tells it to retry or not so that we don't double up on retries when following this "error for retry" pattern

Reply:
makes sense